code/datums/components/healing_touch.dm

```diff
@@ -111,7 +119,20 @@
 
 /// Returns true if the target has a kind of damage which we can heal
 /datum/component/healing_touch/proc/has_healable_damage(mob/living/target)
-	return (target.getBruteLoss() > 0 && heal_brute) || (target.getFireLoss() > 0 && heal_burn) || (target.getStaminaLoss() > 0 && heal_stamina)
+	if (!isnull(valid_biotypes) && !(valid_biotypes & target.mob_biotypes))
+		return FALSE
+	if (target.getStaminaLoss() > 0 && heal_stamina)
+		return TRUE
+	if (!iscarbon(target))
+		return (target.getBruteLoss() > 0 && heal_brute) || (target.getFireLoss() > 0 && heal_burn)
+	var/mob/living/carbon/carbon_target = target
+	for (var/obj/item/bodypart/part in carbon_target.bodyparts)
+		if (!(part.brute_dam && heal_brute) && !(part.burn_dam && heal_burn))
+			continue
+		if (!isnull(required_bodytype) && !(part.bodytype & required_bodytype))
+			continue
```

Comment:
I don't think you need to check every bodypart's bodytype individually, a carbon's bodytype is an aggregate of all limb bodytypes

Likewise you don't need to check individual limbs for brute or burn, you can just use `getXLoss`

Reply:
I think in this case I _might_ have to?
The point is that if you have someone with two damaged flesh arms and one damaged robot leg, it _should_ target them until only the leg remains damaged. I don't think the cumulative bodypart flag would work for this, because it would either refuse to ever heal anyone with a cyborg limb or continue fruitlessly trying to heal damage it can't heal.
So we're checking individual limbs for:
- if they have damage we can heal
- if they are a bodytype we can heal

Reply:
I see